app/scripts/directives/mobileClientServices.js

```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+(function () {
+  angular.module('openshiftConsole').component('mobileClientServices', {
+    controller: [
+      '$filter',
+      'APIService',
+      'DataService',
+      'MobileServicesService',
+      'NotificationsService',
+      MobileClientServicesController
+    ],
+    bindings: {
+      mobileClient: '<',
+      project: '<',
+    },
+    templateUrl: 'views/directives/mobile-client-services.html'
+  });
+
+  function MobileClientServicesController($filter, APIService, DataService, MobileServicesService, NotificationsService) {
+    var ctrl = this;
+    var context = { namespace: ctrl.project };
+    var getErrorDetails = $filter('getErrorDetails');
+    var isServiceInstanceReady = $filter('isServiceInstanceReady');
+    var isMobileService = $filter('isMobileService');
+    ctrl.title = "Mobile Services";
+
+    ctrl.$onInit = function () {
+      DataService.watch(APIService.getPreferredVersion('serviceinstances'), context, function (serviceInstancesData) {
+        var data = serviceInstancesData.by("metadata.name");
+
+        ctrl.services = _.filter(data, function (serviceInstance) {
+          return isMobileService(serviceInstance) && isServiceInstanceReady(serviceInstance);
+        });
+
+        _.each(ctrl.services, function (serviceInstance) {
+          if (!serviceInstance.resourceData) {
+            MobileServicesService.getIcon(serviceInstance).then(function (icon) {
+              serviceInstance.resourceData = {};
+              serviceInstance.resourceData.imageUrl = icon;
```

Comment:
Use `_.set(object, 'a[0].b.c', 4);` In this case `_.set(serviceInstance, 'resourceData.imageUrl', icon)`. Same for others